plugins/com.google.cloud.tools.eclipse.appengine.login/src/com/google/cloud/tools/eclipse/appengine/login/ui/LoginBrowser.java

```diff
@@ -0,0 +1,96 @@
+package com.google.cloud.tools.eclipse.appengine.login.ui;
+
+import java.net.URL;
+
+import org.eclipse.jface.dialogs.Dialog;
+import org.eclipse.jface.dialogs.IDialogConstants;
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.window.IShellProvider;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.browser.Browser;
+import org.eclipse.swt.browser.ProgressAdapter;
+import org.eclipse.swt.browser.ProgressEvent;
+import org.eclipse.swt.browser.TitleEvent;
+import org.eclipse.swt.browser.TitleListener;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Control;
+import org.eclipse.swt.widgets.Shell;
+
+public class LoginBrowser extends Dialog {
+
+  private static final String LOGOUT_URL = "https://www.google.com/accounts/Logout"; //$NON-NLS-1$
+  private static final String SUCCESS_CODE_PREFIX = "Success code="; //$NON-NLS-1$
+
+  private Browser browser;
+  private URL loginUrl;
+
+  private String verificationCode;
+
+  public LoginBrowser(IShellProvider shellProvider, URL loginUrl) {
+    super(shellProvider.getShell());
+    this.loginUrl = loginUrl;
+  }
+
+  public String getVerificationCode() {
+    return verificationCode;
+  }
+
+  @Override
+  protected void configureShell(Shell newShell) {
+    super.configureShell(newShell);
+    newShell.setText("Sign in to Google");  // TODO(chanseok): localize after PR #386 lands.
+  }
+
+  @Override
+  protected Control createDialogArea(Composite parent)
+  {
+    Composite composite = (Composite) super.createDialogArea(parent);
+
+    browser = new Browser(composite, SWT.BORDER);
+    browser.setUrl(loginUrl.toString());
+    browser.addProgressListener(new PageLoadingListener());
```

Comment:
One difference between this and the GPE LoginBrowser: you're not handling extraneous link navigations, when the user clicks on some other link unrelated to the authentication and authorization.  It might be worth having a whitelist of URL prefixes that are ok to go to.